### DISCUSSION TITLE
Create a custom font based on Unifont > tool here !

### DISCUSSION THREAD

**Discussion by stncrn** [olikraus/u8g2#1449] [Show and tell]
Hello, 

Unifont is one of the bitmap fonts offering the widest coverage of Unicode. Several of its subsets are already offered in u8g2 (https://github.com/olikraus/u8g2/wiki/fntgrpunifont)

However, there are some projects where you need characters from several scripts/languages (for example, when making a _now playing_ display/controller for your music player).
And then you need to play with bdfconv and to select the right subsets... I therefore spent quite a long time to write the command line I needed (no joke, it was: `bdfconv -v -f 1 -m "0-127,128-255,256-383,1024-1279,1280-1327,1424-1535,8192-8303,9728-9983,12288-12351,12352-12447,12448-12543,19968,19978,19990,20117,20154,20219,20253,20309,20415,20493,21152,21205,21253,21315,21512,21517,21531,21764,22338,22402,22478,22675,22823,22825,22899,23019,23376,23429,23433,23455,23478,23487,23563,23665,23713,23830,23948,24059,24179,24230,24339,24613,24681,24693,24859,25104,25126,25163,26132,26178,26408,26412,26449,26463,26494,26519,27211,27468,27491,27915,28023,28779,29432,29483,30000,30001,30028,30331,30690,31070,31278,31354,31859,32000,32004,32005,32654,32819,33391,33457,33909,34276,34903,35328,35352,35441,35504,35895,35930,36062,36820,37117,37326,38525,38560,38915,38997,39080,39089,39764" unifont-13.0.06.bdf -o unifont_custom.c -n unifont_custom -d unifont-13.0.06.bdf`)

Something definitely had to be done!

I therefore created a tool to help writing bdfconv parameters, especially with Unifont:
- single webpage for easy portability
- easy 'just click' interface
- support both BMP and SMP (with custom Unifont build)
- show glyphs preview

I let you play around with it:
[https://github.com/stncrn/u8g2-unifont-helper](https://github.com/stncrn/u8g2-unifont-helper)
[https://stncrn.github.io/u8g2-unifont-helper/](https://stncrn.github.io/u8g2-unifont-helper/)

**Comment by olikraus** [comment]
Wow, I am impressed. Nice work.

**Reply by olikraus** [reply]
I have added an entry to the U8g2 FAQ: 
https://github.com/olikraus/u8g2/blob/9ed420b7f27113769d1eb2f7e69ff12d429fb0af/doc/faq.txt#L248-L251


Thanks again
